frontend/src/components/HeaderNav.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component, useState, useRef, useEffect } from 'react';
+import { StyleSheet, View, BackHandler } from 'react-native';
+import { Appbar, Searchbar } from 'react-native-paper';
+import { connect, useDispatch } from 'react-redux';
+import StoreSelector from './StoreSelectorButton';
+// import {  } from './SuggestionList';
+import { headerStates, globalStates } from '../constants/States';
+import { State } from '../constants/Interfaces';
+import { actionTypes } from '../constants/ActionTypes';
+
+interface Props {
+    currState: headerStates
+}
+
+function HeaderNav(props: Props) {
+    let [clearQuery, setClearQuery] = useState(() => false);
+    let [query, setQuery] = useState(() => "");
+
+    const dispatch = useDispatch();
+    
+    useEffect(() => {
```

Comment:
It seems like this effect is just allowing us to react to hardware back presses, but a quick comment above this line would make it easier to figure out what this code does in the future

Reply:
This is mainly for android phones and I implemented it since originally the search page didn't have a back button (therefore couldn't really exit that page). I'll make sure to comment my code more once I feel like everything is finalized to a certain degree